src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -45,6 +45,7 @@
 
 public class DataConverter {
 
+  static boolean useCompactMapEntries = true;
```

Comment:
So this practically means we can't have 2 connectors operating in different modes. 
And, in general, fully static classes that start to become richer in logic could potentially become a problem, both in testing and regular usage. 

This class exposes only one public method (`convertRecord`). I wonder whether it's worth turning it into non-static and instantiating based on the config or punting this issue for the near future. Up to you.